crossbar/router/broker.py

```diff
@@ -171,13 +171,17 @@ def _filter_publish_receivers(self, receivers, publish):
 
         # if "eligible_authid" we only accept receivers that have the correct authid
         if publish.eligible_authid:
+            eligible = set()
             for aid in publish.eligible_authid:
-                receivers = receivers & self._router._authid_to_sessions.get(aid, set())
```

Comment:
I believe my (new) code *is* correcting the "iterative" problem @markope identifies?

That is: I'm iteratively union-ing all the authroles/authid and only *then* intersecting against the possible receivers.